### PR TITLE
StripeConnectAccountService create test

### LINKED ISSUE #564: Add StripeConnectAccountService create test
# Problem

We need to test the `StripeConnectAccountService`'s `create` to create an account with `managed: true` and the country code only.

### PR DESCRIPTION
**Pull request by amyschools**
# What's in this PR?

Adding a test file and specifically a `create` test to StripeConnectAccountService.

## References
Fixes #564 

Progress on: #

### CODE REVIEW

File: test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    test "creates a Stripe Connect Account" do
+      connect_account = insert(:stripe_connect_account)
+
+      connect_account =
+        StripeConnectAccount
+        |> Repo.get(connect_account.id)
+        |> Repo.preload([:stripe_platform_account, :stripe_connect_account])
+
+      created_at = connect_card.created_at
```
Comment (joshsmith) [inline test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs]: This should be `connect_account`/

File: test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+defmodule CodeCorps.StripeService.StripeConnectAccountServiceTest do
+  use ExUnit.Case, async: true
+
+  use CodeCorps.ModelCase
+
+  alias CodeCorps.StripeConnectAccount
+  alias CodeCorps.StripeService.StripeConnectAccountService
+
+  describe "create" do
+      test "creates a Stripe Connect Account" do
```
Comment (joshsmith) [inline test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs]: Would write as `StripeConnectAccount`

File: test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+        attributes = %{"country" => "US", "organization_id" => organization.id}
+
+        {:ok, %CodeCorps.StripeConnectAccount{} = connect_account} =
+                StripeConnectAccountService.create(attributes)
+
+        assert connect_account.country == "US"
+        assert connect_account.organization_id == organization.id
+      end
+    end
```
Comment (joshsmith) [inline test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs:21]: Indent left one.

**Review by joshsmith — COMMENTED** [review]
Quick comments and then you'll be good to go!

**Comment by joshsmith** [comment]
I don't know why these view tests are failing suddenly. @begedin? Some upstream change?

File: test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    test "creates a StripeConnectAccount" do
+      organization = insert(:organization)
+
+      attributes = %{"country" => "US", "organization_id" => organization.id}
+
+      {:ok, %CodeCorps.StripeConnectAccount{} = connect_account} =
+              StripeConnectAccountService.create(attributes)
+
+      assert connect_account.country == "US"
+      assert connect_account.organization_id == organization.id
```
Comment (begedin) [inline test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs:19]: Need to assert it also sets "managed" to "true".

**Review by begedin — DISMISSED** [review]
Requesting a change that needs to be added to fulfil the requirements of the issue.

Outside of that, looks good, and ought to be good to merge.

Ideally, we would also test that the stripe API has been called as well, but we don't have a nice way to do that yet.

I'm not sure we will be creating a stripe account with country code only, though. @JoshSmith, wont it require an email to?

That being said, the answer should not affect this. Merging this one will give us a working test now, and one that will fail nicely and be easily extendible when we continue implementation of managed accounts.s

File: test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs (hunk 4 above)
Comment (amyschools) [reply test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs:19]: I tried putting `"managed" => true` in the attributes and then doing `assert connect_account.managed(true) == :managed` but it looks like managed is still false. Is there some other way to set it to true?

**Comment by joshsmith** [comment]
@begedin sorry, the reason I was asking your thoughts on this was because of the failures on Circle unrelated to @amyschools's changes.

File: test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs (hunk 4 above)
Comment (joshsmith) [reply test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs:19]: @amyschools in `CodeCorps.StripeTesting.Account` change `managed` to `true`.

Comment (joshsmith) [reply test/lib/code_corps/stripe_service/stripe_connect_account_service_test.exs:19]: That is a mock of Stripe.

**Comment by begedin** [comment]
From https://github.com/vt-elixir/ja_serializer/blob/master/CHANGELOG.md

>Bugfixes
>  Add json api version to error responses. (@nmcalabroso)

So yeah, we need to add the version property to our error view tests.

Basically, our error view now adds this bit when rendering:
```
 "jsonapi" => %{"version" => "1.0"}
```

So we need our test to expect that bit as well:

In `error_view_test`, this:

```Elixir
assert render(CodeCorps.ErrorView, "404.json-api", []) ==
           %{"errors" => [%{status: 404, title: "404 Resource not found", id: "NOT_FOUND"}]}

```

Needs to become

```Elixir
rendered_json =  render(CodeCorps.ErrorView, "404.json-api", [])

expected_json = %{
  "errors" => [%{status: 404,  title: "404 Resource not found", id: "NOT_FOUND"}],
   "jsonapi" => %{"version" => "1.0"}
}
assert rendered_json == expected_json
```

A similar thing needs to happen with the `500.json-api` test.

**Comment by joshsmith** [comment]
@amyschools the tests are failing for a different reason now.

File: test/views/error_view_test.exs [hunk 5]
```diff
@@ -5,13 +5,23 @@ defmodule CodeCorps.ErrorViewTest do
...
+    assert rendered_json == expected_json
   end
 
   test "renders 500.json-api" do
-    assert render(CodeCorps.ErrorView, "500.json-api", []) ==
-           %{"errors" => [%{status: 500, title: "500 Internal server error", id: "INTERNAL_SERVER_ERROR"}]}
+    rendered_json =  render(CodeCorps.ErrorView, "500.json-api", [])
+
+    expected_json = %{
+      "errors" => [%{status: 500,  title: "500 Internal Server Error", id: "NOT_FOUND"}],
```
Comment (joshsmith) [inline test/views/error_view_test.exs]: The `id` on this was changed incorrectly, as was the title.

**Comment by joshsmith** [comment]
@amyschools looks good! Can you squash?

**Review by joshsmith — APPROVED** [review]

**Comment by joshsmith** [comment]
Merge at will! 🙌

**Pull request merged**